test/integration/Test/Integration/Framework/DSL.hs

```diff
@@ -88,6 +239,13 @@ infixr 5 </>
 (</>) :: ToHttpApiData a => Text -> a -> Text
 base </> next = mconcat [base, "/", toQueryParam next]
 
+unsafeMkAddressPoolGap :: Word8 -> AddressPoolGap
+unsafeMkAddressPoolGap g = case gap of
+    Right a -> a
+    Left _ -> defaultAddressPoolGap
```

Comment:
let's not silently do this but instead, throw hard and crash the tests. If something wrong is provided here, we want to know it otherwise, it may get very confusing to debug this!

Reply:
yeah, I only created it to use in the lens for `addressPoolGap` in the setter (because the _safe constructor `mkAddressPoolGap` did not type check as it returns Either _ AddressPoolGap):
```haskell
    _set :: HasType (ApiT AddressPoolGap) s => (s, Word8) -> s
    _set (s, v) = set typed (ApiT $ unsafeMkAddressPoolGap v) s
```
I can probably remove `unsafeMkAddressPoolGap` and do smth like this, since the setter will never be used most probably:
```haskell
    --FIXME setter is dummy and always sets defaultAddressPoolGap
    _set :: HasType (ApiT AddressPoolGap) s => (s, Word8) -> s
    _set (s, _) = set typed (ApiT defaultAddressPoolGap) s
```

Reply:
Or simply replace the `defaultAddressPoolGap` with an appropriate invariant / `error` call. In this particular context, it's okay.

Reply:
> Or simply replace the defaultAddressPoolGap with an appropriate invariant / error call. 

Would you mind giving an example please? :)

Reply:
```hs
unsafeMkAddressPoolGap :: Word8 -> AddressPoolGap
unsafeMkAddressPoolGap g = case gap of
    Right a -> a
    Left _ -> error $ "unsafeMkAddressPoolGap: bad argument: " <> show g
```